### ISSUE TITLE
showDialog method should return data.

### ISSUE THREAD

**Issue by detroxx92** [bdlukaa/fluent_ui#2]
When using showDialog i want to return data from the dialog to the "caller"-method that called showDialog.
The showDialog returns Future<void> but should be Future<T?>showDialog<T>(...).

Example "Caller":
````
showDialog(
      context: context,
      builder: (_) => LoginDialog(),
    ).then((result) => {
      print("Got result:"),
      print(result),
    });
````
Here i need the result to not be void.

Example LoginDialog:
````
onPressed: () => {
  Navigator.pop(context, {userController.text, passwordController.text})
},
````
I am passing the text values of two TextBoxes to the "caller".

**Comment by detroxx92** [comment]
I wanted to create a pull request, but there are no other branches or i am not used to github at the time 😅

**Comment by bdlukaa** [comment]
Hey, thanks for reporting this issue. This will be fixed in the next version.

**Comment by bdlukaa** [comment]
Fixed in v[1.2.3](https://pub.dev/packages/fluent_ui)
